include/swift/AST/DiagnosticsSIL.def

```diff
@@ -173,6 +173,9 @@ ERROR(immutable_value_passed_inout,none,
 ERROR(assignment_to_immutable_value,none,
       "immutable value '%0' may not be assigned to",
       (StringRef))
+ERROR(mutating_protocol_witness_method_on_let_constant,none,
+      "Cannot perform mutating instruction: '%0' is a 'let' constant",
+      (StringRef))
```

Comment:
I would reuse the existing diagnostic for mutating operations on 'let' constants instead of adding a new one.